api/src/main/java/jakarta/enterprise/invoke/Invoker.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022 Red Hat and others
+ *
+ * This program and the accompanying materials are made available under the
+ * Apache Software License 2.0 which is available at:
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jakarta.enterprise.invoke;
+
+/**
+ * Allows indirectly invoking a method that belongs to a managed bean (the <em>target method</em>).
+ * To invoke the method, the caller must provide all the arguments that the target method accepts,
+ * as well as the instance on which the target method is to be invoked, if it is not {@code static}.
+ * <p>
+ * Whenever a direct invocation of a method is a business method invocation, an indirect invocation
+ * of that method through an invoker is also a business method invocation.
+ * <p>
+ * Invoker implementations must be thread-safe. It is possible to use a single invoker instance
+ * to perform multiple independent invocations of the target method, possibly on different instances
+ * and with different arguments.
+ *
+ * <h2>Example</h2>
+ *
+ * To illustrate how an invoker works, let's take a look at an example. Say we have
+ * the following bean with a method we want to invoke:
+ *
+ * <pre>
+ * class MyService {
+ *     String hello(String name) {
+ *         return "Hello " + name + "!";
+ *     }
+ * }
+ * </pre>
+ *
+ * The CDI container allows building an invoker for this method. Such invoker is equivalent
+ * to the following class:
+ *
+ * <pre>
+ * class TheInvoker implements Invoker&lt;MyService, String&gt; {
+ *     String invoke(MyService instance, Object[] arguments) {
+ *         String argument = (String) arguments[0];
+ *         String result = instance.hello(argument);
+ *         return result;
+ *     }
+ * }
+ * </pre>
+ *
+ * Therefore, calling {@code invoker.invoke(myService, new Object[] {"world"})}
+ * will return {@code "Hello world!"}.
```

Comment:
This example is helpful because it shows how the arguments to `invoke` are used, but at the same time it's a bit misleading because there's no information about why you would want to do this.

I can see a few options:
- We could state that this is a _basic_ invoker and that invokers with more complex behaviour can be created with `InvokerBuilder`.
- We could remove the example altogether and direct users to the examples on `InvokerBuilder`
- We could remove the "equivalent class" part and just give a more complete example of creating an invoker with `InvokerBuilder` and then calling it.

Reply:
Good point, this is indeed a simple invoker with no extra features, which is not exactly apparent in this example. I'll try to expand it.

Reply:
Expanded the example, which prompted moving the _Obtaining an invoker_ section above the _Example_.